src/pybind/mgr/nfs/export.py

```diff
@@ -37,7 +49,14 @@ def cluster_check(
         """
         This method checks if cluster exists
         """
-        if kwargs['cluster_id'] not in available_clusters(export.mgr):
+        try:
+            clusters = set(available_clusters(export.mgr))
+        except NoOrchestrator:
+            clusters = nfs_rados_configs(export.mgr.rados)
```

Comment:
Is there any reason to not always check the nfs rados config (either on its own or along with available_clusters depending on if the user is using an orchestrator backend)? Does it cause issues if you check nfs rados configs when there is a backend?

Reply:
I figured it was better to continue using the previous approach when possible, in case there are any gotchas that I missed from the rados based approach. In theory they are equivalent though.

Reply:
My tendency would be towards option 3 or 4 (1 is also okay, not a huge fan of 2). What kind of differences do you think we'd be seeing that would potentially be logged?

Reply:
There shouldn't really be any state from the orchestrator right? It's possible I've gotten lost in the code a bit, but from my understanding, the orchestrator info determines which rados pool to use for the configs, and possibly the name of the config object. And I think both of those are determined by the NFS cluster-id.

If that's right, IMO, the best approach would be to fail out if the orchestrator is enabled and the user passes a cluster-id that doesn't exist according to the orchestrator. Then, if there is no orchestrator, assume the cluster-id passed by the user is the name of the pool. And if the pool doesn't exist, fail out.

Reply:
FWIW I think @BlaineEXE is mostly describing option 1. Option 1 uses the only the data from orchestrator to validate the cluster_id, when an orchestrator is enabled. In option 1 we fall back to examining what rados namespaces exist in the pool to determine if the cluster_id matches an existing namespace.

All the options should be the same in theory. Options 3 and 4 should ideally be the same in practice too, but I've had too many real-world experiences when you have two possible sources for data disagree. :-)

I personally lean towards option 1, but that should not be a surprise as that's what I have in the code currently. :-)